src/Engine/Manager/VRManager.cpp

```diff
@@ -132,8 +130,54 @@ void VRManager::Submit(vr::Hmd_Eye eye, vr::Texture_t* texture) const {
         Log() << "Unable to submit texture to hmd: " << eError << "\n";
 }
 
-glm::mat4 VRManager::ConvertMatrix(const vr::HmdMatrix34_t& mat)
-{
+float VRManager::GetScale() const {
+    return scale;
+}
+
+void VRManager::SetScale(float scale) {
+    this->scale = scale;
+}
+
+bool VRManager::GetInput(vr::EVRButtonId buttonID) {
+    for (vr::TrackedDeviceIndex_t unDevice = 0; unDevice < vr::k_unMaxTrackedDeviceCount; unDevice++) {
+        vr::VRControllerState_t controllerState;
+        if (vrSystem->GetControllerState(unDevice, &controllerState, sizeof(controllerState))) {
+            pressedTrackedDevice[unDevice] = controllerState.ulButtonPressed == 0;
+            if (controllerState.ulButtonPressed & vr::ButtonMaskFromId(buttonID))
+                return true;
+        }
+    }
+    return false;
+}
+
+Component::VRDevice* VRManager::CreateVRDevice() {
+    return vrDevices.Create();
+}
+
+Component::VRDevice* VRManager::CreateVRDevice(const Json::Value& node) {
+    Component::VRDevice* vrDevice = vrDevices.Create();
+
+    // Load values from Json node.
+    std::string type = node.get("type", "controller").asString();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'SetScale' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)